compiler/src/dmd/link.d

```diff
@@ -1433,18 +1436,26 @@ void parseLinkerOutput(const(char)[] linkerOutput, ErrorSink eSink)
 
     bool missingCSymbols = false;
     bool missingDsymbols = false;
+    bool missingDruntimeSymbols = false;
+    bool missingPhobosSymbols = false;
     bool missingMain = false;
 
     void missingSymbol(const(char)[] name, const(char)[] referencedFrom)
     {
         import core.demangle: demangle;
         if (name.startsWith("__D"))
             name = name[1 .. $]; // MS LINK prepends underscore to the existing one
-        auto sym = demangle(name);
+
+        bool alreadyDemangled = !!findSplit(name, ".");
+        auto sym = alreadyDemangled ? name : demangle(name);
 
         if (sym == "main")
             missingMain = true;
-        else if (sym != name)
+        else if (sym.startsWith("core."))
+            missingDruntimeSymbols = true;
+        else if (sym.startsWith("std."))
```

Comment:
might as well future proof this 
```diff
-        else if (sym.startsWith("std."))
+        else if (sym.startsWith("std.") || sym.startsWith("phobos."))
```

Reply:
The phobos package isn't shipped and we don't know how it's going to link yet, there was talk of it being a 'header only' / source package, in which case the error wouldn't apply. There's also talk about 'platform' and 'kernel' packages right now, so I'd rather cross that bridge when we get there than trying to predict the future.